### PR TITLE
Event engine endpoint read and write API changes

### PR DESCRIPTION
**Pull request by Vignesh2208**
Adding a ReadArgs and WriteArgs to Endpoint Read/Write API  to allow gRPC to control the framing structure of byte streams exchanged between endpoints

@drfloob

### CODE REVIEW

File: include/grpc/event_engine/event_engine.h [hunk 1]
```diff
@@ -176,7 +197,7 @@ class EventEngine {
     /// statuses to \a on_writable. For example, callbacks might expect to
     /// receive CANCELLED on endpoint shutdown.
     virtual void Write(std::function<void(absl::Status)> on_writable,
-                       SliceBuffer* data) = 0;
+                       SliceBuffer* data, const WriteArgs* args) = 0;
```
Comment (drfloob) [inline include/grpc/event_engine/event_engine.h:200]: What about making this default to `nullptr`? I think that's the most common case based on looking at `grpc_endpoint_write` usage.

File: include/grpc/event_engine/event_engine.h [hunk 2]
```diff
@@ -156,7 +168,16 @@ class EventEngine {
...
-                      SliceBuffer* buffer) = 0;
+                      SliceBuffer* buffer, const ReadArgs* args) = 0;
+    /// A struct representing optional arguments that may be provided to an
+    /// Event-Engine Endpoint Write API call.
+    ///
+    /// Passed as argument to an Endpoint \a Write
+    struct WriteArgs {
+      // Represents private information that may be passed by gRPC for
+      // select endpoints expected to be used only within google.
+      void* google_specific;
```
Comment (drfloob) [inline include/grpc/event_engine/event_engine.h:179]: `= nullptr`

**Review by drfloob — COMMENTED** [review]
Thanks for doing this, Vignesh.

File: include/grpc/event_engine/event_engine.h (hunk 2 above)
Comment (Vignesh2208) [reply include/grpc/event_engine/event_engine.h:179]: Done.

Comment (Vignesh2208) [reply include/grpc/event_engine/event_engine.h:179]: Done.

File: include/grpc/event_engine/event_engine.h (hunk 1 above)
Comment (Vignesh2208) [reply include/grpc/event_engine/event_engine.h]: Done.

File: include/grpc/event_engine/event_engine.h (hunk 2 above)
Comment (drfloob) [reply include/grpc/event_engine/event_engine.h:179]: This does not appear to be done.

**Review by drfloob — APPROVED** [review]

**Pull request merged**